### PR TITLE
UI: Allow navigate to list from View Secret card

### LINKED ISSUE #20563: Kv2 inconsistent path segment list/show
When I input a path on kv2 web ui box like the following:

 ccloud/apikey/key1

I get redirected to the following url:

 https://my-vault.com/ui/vault/secrets/kv2/list/ccloud/apikey/key1
 
but if I input a shorter path based on the above like the following:

 ccloud/

I get redirected to the following url with a change from "list" to "show" on the segment before the secret name and a permissions error:

 https://my-vault.com/ui/vault/secrets/kv2/show/ccloud/

It is true that there is no permissions on a show segment, there is no permission "show", it should actually be a "list" segment.
Now if I manually change the url from "show" to "list" on the browser it works and there is no permissions error!

 https://my-vault.com/ui/vault/secrets/kv2/show/ccloud/ ->  https://my-vault.com/ui/vault/secrets/kv2/list/ccloud/

1. Authenticate in Vault Web UI
2. Select Secrets tab
3. Select KV2 secrets engine
4. Input an existing secret on the input box

The input box of kv2 should actually use a "list" path segment, not a "show" because that's the name of the permission to see the contents on the path, i.e., "list"!

Vault Server version: 1.12.1

Policy:

path "kv2/" {
    capabilities = [ "list" ]
}
path "kv2/data" {
    capabilities = [ "list" ]
}
path "kv2/data/ccloud/" {
    capabilities = [ "list" ]
}
path "kv2/data/ccloud/*" {
    capabilities = [ "list", "read" ]
}
path "kv2/metadata/ccloud/*" {
    capabilities = [ "list" ]
}
path "kv2/metadata/ccloud/*" {
    capabilities = [ "read" ]
}

### PR DESCRIPTION
**Pull request by hashishaw**
Before this fix, the View Secret card that shows when no `LIST` permissions on a KV engine would only navigate to the show secret page. 
View secret card on KV list view

With this update, if the input ends in a slash the button will update to say "View list" and it will take you to the correct subdirectory in KV: 
Button text updates when input ends with /
From the linked list page, it prefills the textbox and the user can type in the rest of the path

I also changed the way the form is triggered, so that it will trigger both on button click and "Enter" key (like a normal form)

Fixes #20563

### CODE REVIEW

File: ui/app/templates/vault/cluster/secrets/backend/list.hbs
Comment (hellobontempo) [inline ui/app/templates/vault/cluster/secrets/backend/list.hbs:1]: Do we want to update the subtext here slightly to mention the trailing slash like we did in the new engine? https://github.com/hashicorp/vault/blob/799c4319a9cc581230390c7842a63b2824ffae0f/ui/lib/kv/addon/components/page/list.hbs#L30

File: ui/app/components/get-credentials-card.js [hunk 1]
```diff
@@ -32,18 +33,38 @@ export default class GetCredentialsCard extends Component {
...
   @tracked secret = '';
 
+  constructor() {
+    super(...arguments);
+    this.secret = this.args?.initialValue || '';
+  }
+
+  get buttonText() {
+    if (this.args.type === 'secret') {
+      if (this.secret.endsWith('/')) {
```
Comment (hellobontempo) [inline ui/app/components/get-credentials-card.js:43]: I didn't know `endsWith()` was a thing!! So cool 🤩

**Review by hellobontempo — APPROVED** [review]
🚢  Thanks for tackling!

**Comment by hashishaw** [comment]
Can only backport as far as 1.13 because the base changes in `GetCredentialsCard` are not present in 1.12

**Pull request merged**